airflow/models/dag.py

```diff
@@ -745,10 +745,18 @@ def tasks(self, val):
     def task_ids(self) -> List[str]:
         return list(self.task_dict.keys())
 
+    @property
+    def task_group_dict(self) -> Dict[str, "TaskGroup"]:
+        return {k: v for k, v in self._task_group.get_task_group_dict().items() if k is not None}
```

Comment:
Since `.task_group` does the same

```suggestion
        return {k: v for k, v in self.task_group.get_task_group_dict().items() if k is not None}
```